lci_strategic_plugin/src/lci_strategic_plugin.cpp

```diff
@@ -369,82 +369,242 @@ void LCIStrategicPlugin::handleGreenSignalScenario(const cav_srvs::PlanManeuvers
 {
   //Requested maneuver needs to be modified to meet remaining_dist req
   //by trying to get close to the target_speed and remaining_time as much as possible
-  TrajectoryParams params;
-
-  params.is_algorithm_successful = false;
-  params.case_num = CASE_1;
+  TrajectoryParams return_params;
+  TrajectoryParams traj_upper;
+  TrajectoryParams traj_lower;
 
   ROS_DEBUG_STREAM("HANDLE_LAST_RESORT_CASE: Starting...");
-  double modified_remaining_time;
+  double starting_downtrack = traffic_light_downtrack - remaining_downtrack;
+  double modified_remaining_time_upper; // upper meaning downtrack vs time trajectory is curved upwards
+  double modified_remaining_time_lower; // upper meaning downtrack vs time trajectory is curved lower
+  double modified_departure_speed_upper;
+  double modified_departure_speed_lower;
+  bool calculation_success_upper = true; // identifies places in codes where calculation can be invalid such as negative distance
+
+  // TODO change light_controlled_intersection_tactical_plugin...
+
+  // the upper ET
+  // accel case
+  traj_upper.t0_ = current_time;
+  traj_upper.v0_ = starting_speed;
+  traj_upper.x0_ = starting_downtrack;
+  traj_upper.is_algorithm_successful = false;
+  traj_upper.case_num = CASE_1;
+
+  if (departure_speed > starting_speed)
+  {
+    if ((pow(departure_speed,2) - pow(starting_speed,2))/(2*max_comfort_accel_) >= remaining_downtrack)
+    {
+      ROS_DEBUG_STREAM("Failure case upper 1");
+      
+      modified_departure_speed_upper = sqrt(pow(starting_speed, 2) + (2 * max_comfort_accel_ * remaining_downtrack));
+      modified_remaining_time_upper = (modified_departure_speed_upper - starting_speed) / max_comfort_accel_;
 
-  if (starting_speed <= departure_speed)
-    modified_remaining_time = (sqrt(pow(starting_speed, 2) + (2 * max_comfort_accel_ * remaining_downtrack)) - starting_speed)/ max_comfort_accel_;
-  else
-    modified_remaining_time = (sqrt(pow(starting_speed, 2) + (2 * max_comfort_decel_ * remaining_downtrack)) - starting_speed) / max_comfort_decel_;
+      traj_upper.t1_ = current_time + modified_remaining_time_upper;
+      traj_upper.v1_ = modified_departure_speed_upper;
+      traj_upper.a1_ = max_comfort_accel_;
+      traj_upper.x1_ = traffic_light_downtrack;
 
-  if (starting_speed <= departure_speed)
-  {
-    params.a1_ = max_comfort_accel_;
-    params.v1_ = sqrt(pow(starting_speed, 2) + (2 * max_comfort_accel_ * remaining_downtrack));
+      traj_upper.t2_ = traj_upper.t1_;
+      traj_upper.v2_ = traj_upper.v1_;
+      traj_upper.a2_ = traj_upper.a1_;
+      traj_upper.x2_ = traj_upper.x1_;
+
+      traj_upper.modified_departure_speed = modified_departure_speed_upper;
+      traj_upper.modified_remaining_time = modified_remaining_time_upper;
+    }
+    else // NOTE: most likely will not happen as it would have happened at trajectory smoothing part
+    { 
+      ROS_DEBUG_STREAM("Failure case upper 2");
```

Comment:
It is not technically a failure case. It just means that the vehicle cannot reach to departure speed which is ok. You can say, HandleFailureCase -> Upper Trajectory -> Actual speed greater than desired departure speed.